src/hotspot/os/linux/cgroupV1Subsystem_linux.cpp

```diff
@@ -39,6 +39,13 @@
  * on the contents of the mountinfo and cgroup files.
  */
 void CgroupV1Controller::set_subsystem_path(char *cgroup_path) {
+  if (_cgroup_path != nullptr) {
+    os::free(_cgroup_path);
+  }
+  if (_path != nullptr) {
+    os::free(_path);
```

Comment:
Please set _path to nullptr, the dangling pointer makes me nervous.

Reply:
You free _path here, but it is still not nullptr. You then rely on all three code paths below to reset _path to a sensible value. That is brittle and may bitrot. It would be cleaner to add a _path = nullptr right after the free, to prevent the old invalif pointer from being accidentally used.

Reply:
Ah, makes sense.

Reply:
Done now.